x-pack/plugins/cases/public/components/user_actions/use_user_actions_handler.tsx

```diff
@@ -0,0 +1,171 @@
+};
+
+export const useUserActionsHandler = ({
+  fetchUserActions,
+  updateCase,
+}: UseUserActionsHandlerArgs): UseUserActionsHandler => {
+  const { detailName: caseId, subCaseId } = useCaseViewParams();
+  const { clearDraftComment, draftComment, hasIncomingLensState, openLensModal } =
+    useLensDraftComment();
+  const handlerTimeoutId = useRef(0);
+  const { isLoadingIds, patchComment } = useUpdateComment();
+  const [selectedOutlineCommentId, setSelectedOutlineCommentId] = useState('');
+  const [manageMarkdownEditIds, setManageMarkdownEditIds] = useState<string[]>([]);
+  const commentRefs = useRef<
+    Record<string, AddCommentRefObject | UserActionMarkdownRefObject | undefined | null>
+  >({});
+
+  const handleManageMarkdownEditId = useCallback(
+    (id: string) => {
+      clearDraftComment();
+      setManageMarkdownEditIds((prevManageMarkdownEditIds) =>
+        !prevManageMarkdownEditIds.includes(id)
+          ? prevManageMarkdownEditIds.concat(id)
+          : prevManageMarkdownEditIds.filter((myId) => id !== myId)
+      );
+    },
+    [clearDraftComment]
+  );
+
+  const handleSaveComment = useCallback(
+    ({ id, version }: { id: string; version: string }, content: string) => {
+      patchComment({
+        caseId,
+        commentId: id,
+        commentUpdate: content,
+        fetchUserActions,
+        version,
+        updateCase,
+        subCaseId,
+      });
+    },
+    [caseId, fetchUserActions, patchComment, subCaseId, updateCase]
+  );
+
+  const handleOutlineComment = useCallback(
+    (id: string) => {
+      const moveToTarget = document.getElementById(`${id}-permLink`);
+      if (moveToTarget != null) {
+        const yOffset = -120;
+        const y = moveToTarget.getBoundingClientRect().top + window.pageYOffset + yOffset;
+        window.scrollTo({
+          top: y,
+          behavior: 'smooth',
+        });
+
+        if (id === 'add-comment') {
+          moveToTarget.getElementsByTagName('textarea')[0].focus();
+        }
+      }
+
+      window.clearTimeout(handlerTimeoutId.current);
+      setSelectedOutlineCommentId(id);
+
+      handlerTimeoutId.current = window.setTimeout(() => {
+        setSelectedOutlineCommentId('');
+        window.clearTimeout(handlerTimeoutId.current);
+      }, 2400);
+    },
+    [handlerTimeoutId]
+  );
+
+  const handleManageQuote = useCallback(
+    (quote: string) => {
+      const ref = commentRefs?.current[NEW_COMMENT_ID];
+      if (isAddCommentRef(ref)) {
+        ref.addQuote(quote);
+      }
+
+      handleOutlineComment('add-comment');
+    },
+    [handleOutlineComment]
+  );
+
+  const handleUpdate = useCallback(
+    (newCase: Case) => {
+      updateCase(newCase);
+      fetchUserActions();
+    },
+    [fetchUserActions, updateCase]
+  );
+
+  useEffect(() => {
+    if (draftComment?.commentId) {
+      setManageMarkdownEditIds((prevManageMarkdownEditIds) => {
+        if (
+          ![NEW_COMMENT_ID].includes(draftComment?.commentId) &&
```

Comment:
nit: do we need the `.includes` here? Can we do `NEW_COMMENT_ID !== draftComment?.commentId`?

Reply:
Nope. Old code 🙂